### PR TITLE
Also reset rotation when zooming to an extent

### PR DESCRIPTION
**Pull request by marcjansen**
Title says it all; this allows people to also reset the rotation when the zoom to the start extent.

I also added docs, some `TODO` and harmonized the look of the source code.

Please review.

### CODE REVIEW

File: src/view/button/ZoomToExtent.js [hunk 1]
```diff
@@ -48,43 +54,59 @@ Ext.define('BasiGX.view.button.ZoomToExtent', {
...
-            var targetCenter = this.getCenter();
-            var targetResolution = this.getResolution();
-            var targetZoom = this.getZoom();
+
+            var targetCenter = me.getCenter();
+            var targetResolution = me.getResolution();
+            var targetRotation = me.getRotation();
+            var targetZoom = me.getZoom();
+
+            // Create the animation with their respectiove start values:
```
Comment (annarieger) [inline src/view/button/ZoomToExtent.js]: Typo `respective`

**Comment by annarieger** [comment]
Thanks for adapting, looks good to me :pineapple: 

Please fix typo and merge at will.

**Comment by dnlkoch** [comment]
Great!

**Comment by marcjansen** [comment]
Thanks for the reviews and the merge :wink:

**Pull request merged**